cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/lib/table/hbase/HBaseMetricsTable.java

```diff
@@ -95,11 +96,19 @@ public HBaseMetricsTable(DatasetContext datasetContext, DatasetSpecification spe
     this.columnFamily = TableProperties.getColumnFamilyBytes(spec.getProperties());
   }
 
-  private void initializeV3Vars(DatasetSpecification spec) {
+  private void initializeV3Vars(CConfiguration cConf, DatasetSpecification spec) {
     boolean isV3Table = spec.getName().contains("v3");
     this.scanExecutor = null;
     this.rowKeyDistributor = null;
     if (isV3Table) {
+      // If v3 table is already created, we will ignore splits value from cConf to avoid modifying splits after creation
```

Comment:
It's ok to expose a method to get the spec (or only the number of splits). MetricsTable is an internal dataset and only used by CDAP.